### PR TITLE
Lesson 3 Hometask by Philipp Skrupskiy

### PR DESCRIPTION
**Pull request by skrphv**
Please, check my lesson3_hometask resolving.

### CODE REVIEW

File: Lesson3Hometask/task1_stack/main.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        else if (bracket == ')') {
+            if (!text_stack.empty()) {
+                text_stack.pop();
+            }
+            else {
+                return false;
+            }
+        }
+        else {
+            throw std::logic_error("Bad symbol!");
```
Comment (smay1613) [inline Lesson3Hometask/task1_stack/main.cpp:24]: Wow, nice!

File: Lesson3Hometask/task1_stack/main.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+
+    if (text_stack.empty()) {
+        return true;
+    }
+
+    return false;
+    //NOTE: я бы использовал тут просто int, и проверял бы "int < 0" и "int == 0"
```
Comment (smay1613) [inline Lesson3Hometask/task1_stack/main.cpp:33]: Returning int is not a good idea.

File: Lesson3Hometask/task1_stack/main.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+int main()
+{
+    std::string valid_str1 {"()"};
+    std::string valid_str2 {"((()()())()())"};
+
+    std::string invalid_str1 {")("};
+    std::string invalid_str2 {"())("};
+
+    try {
+        assert (is_valid (valid_str1));
```
Comment (smay1613) [inline Lesson3Hometask/task1_stack/main.cpp:46]: Try with gtest please.

**Review by smay1613 — COMMENTED** [review]
Great work!

File: Lesson3Hometask/task1_stack/main.cpp (hunk 2 above)
Comment (skrphv) [reply Lesson3Hometask/task1_stack/main.cpp:33]: I mean, instead of std::stack;

**Pull request closed without merging**